solr/modules/llm/src/java/org/apache/solr/llm/texttovector/update/processor/TextToVectorUpdateProcessor.java

```diff
@@ -0,0 +1,100 @@
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.solr.llm.texttovector.update.processor;
+
+import org.apache.solr.common.SolrException;
+import org.apache.solr.common.SolrInputDocument;
+import org.apache.solr.common.SolrInputField;
+import org.apache.solr.llm.texttovector.model.SolrTextToVectorModel;
+import org.apache.solr.llm.texttovector.store.rest.ManagedTextToVectorModelStore;
+import org.apache.solr.request.SolrQueryRequest;
+import org.apache.solr.update.AddUpdateCommand;
+import org.apache.solr.update.processor.UpdateRequestProcessor;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+import java.io.IOException;
+import java.lang.invoke.MethodHandles;
+import java.util.ArrayList;
+import java.util.List;
+
+
+class TextToVectorUpdateProcessor extends UpdateRequestProcessor {
+    private static final Logger log = LoggerFactory.getLogger(MethodHandles.lookup().lookupClass());
+
+    private final String inputField;
+    private final String outputField;
+    private final String model;
+    private SolrTextToVectorModel textToVector;
+    private ManagedTextToVectorModelStore modelStore = null;
+
+    public TextToVectorUpdateProcessor(
+            String inputField,
+            String outputField,
+            String model,
+            SolrQueryRequest req,
+            UpdateRequestProcessor next) {
+        super(next);
+        this.inputField = inputField;
+        this.outputField = outputField;
+        this.model = model;
+        this.modelStore = ManagedTextToVectorModelStore.getManagedModelStore(req.getCore());
+    }
+
+    /**
+     * @param cmd the update command in input containing the Document to process
+     * @throws IOException If there is a low-level I/O error
+     */
+    @Override
+    public void processAdd(AddUpdateCommand cmd) throws IOException {
+        this.textToVector = modelStore.getModel(model);
+        if (textToVector == null) {
+            throw new SolrException(
+                    SolrException.ErrorCode.BAD_REQUEST,
+                    "The model requested '"
+                            + model
+                            + "' can't be found in the store: "
+                            + ManagedTextToVectorModelStore.REST_END_POINT);
+        }
+
+        SolrInputDocument doc = cmd.getSolrInputDocument();
+        SolrInputField inputFieldContent = doc.get(inputField);
+        if (!isNullOrEmpty(inputFieldContent, doc, inputField)) {
+            String textToVectorise = inputFieldContent.getValue().toString();//add null checks and
+            float[] vector = textToVector.vectorise(textToVectorise);
```

Comment:
So text-to-vector in the search case is ephemeral i.e. in case of errors or timeouts the user gets an error or exception back and they may or may not choose to retry.

For text-to-vector in the update case, might some users prefer to index only documents with (all the) vectors and others would rather have the document indexed even if some vectors are missing? (I assume but don't know for sure that the `vectorise` call might throw an exception in certain circumstances and if it's not caught that might fail the indexing for the entire document.)

Reply:
I was chatting with @iamsanjay this morning, and I was expounding on the thought that a lot of folks might want to first index the document with just the core text/string/numbers, and then, since enrichment is SLOW, come back with a streaming expression and do things like vectorization, and an atomic update..  that way you pump your data in as fast as possible, and then enrich at your leisure...    This model constrains your indexing speed to your vectorization speed.  Not saying we should have the update request processor approach, but I also want to see a "out of band" vectorization process.

Reply:
1) @cpoerschke : I double checked and the langchain4j library 'embed' method (that's used in our 'vecctorise' method), doesn't return any exception, but I gree we should investigate what happens if that request fails (my best guess is we get an empty vector or null, I'll add that to tests)

2) @epugh : given that 'update.chain' is a parameter, if you configure a chain with no vector enrichment and a chain with vector enrichment, what prevents you from first index using the 'no vectors' chain and then slowly updating the index with atomic updates that add vectors (using the vector-chain)? We should double check and add to the documentation once we consolidate the code, what do you think?